### PR TITLE
Add FullCalendar Widget Implementation with Nostr Integration

### LINKED ISSUE #2: BOUNTY: Coverage for widget library bindings and options
We are using several open source libraries to provide widget components. The WASM bindings and options are very incomplete and coverage needs to be improved. 

Coverage of libraries should at least include: 

- All initial option object configurations
- All event types
- Demo of Yew functional components showcasing Nostr integration

### PR DESCRIPTION
**Pull request by rahat2134**
Fix #2 - Full calender widget
## Changes ##

Added FullCalendar widget bindings and components
Implemented Nostr integration for calendar events
Added event creation and management functionality
Added toast notifications for user feedback

## Features ##

Full calendar view with month/week/day layouts
Event creation via click & drag
Event storage using Nostr notes (kind: 31924)
Real-time event synchronization via Nostr relay
Toast notifications for user actions

## Testing Done ##

Calendar view rendering
Event creation & display
View switching (month/week/day)
Nostr event broadcasting
Toast notifications

## Dependencies Added ##

FullCalendar CSS and JS (via CDN)

### CODE REVIEW

File: src/widgets/full_calendar/test_component.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+use wasm_bindgen::JsValue;
+use crate::widgets::toastify::ToastifyOptions;
+
+#[function_component(FullCalendarTest)]
+pub fn calendar_test() -> Html {
+    let relay_ctx = use_context::<NostrProps>().expect("No relay context found");
+    let events = use_state(Vec::new);
+    {
+        let relay_ctx = relay_ctx.clone();
+        use_effect_with((), move |_| {
```
Comment (42Pupusas) [inline src/widgets/full_calendar/test_component.rs:18]: Need to emit a `NostrSubscription` to the `relay_ctx` to listen to the necessary evnt kinds, currently not receiving anything:

image

File: src/widgets/full_calendar/component.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    // Update events when props change
+    {
+        let calendar = calendar.clone();
+        let events = props.events.clone();
+        
+        use_effect_with(events, move |events| {
+            if let Some(calendar_instance) = (*calendar).clone() {
+                calendar_instance.clear_events();
+                for event in events {
+                    calendar_instance.add_or_replace_event(event.clone());
```
Comment (42Pupusas) [inline src/widgets/full_calendar/component.rs]: please make sure no unhandled results for `add_or_replace_event` uses

File: src/widgets/full_calendar/full_calendar.rs [hunk 3]
```diff
@@ -33,32 +33,73 @@ extern "C" {
...
         }
     }
-    pub fn add_or_replace_event(&self, event: FullCalendarEvent) {
-        if let Some(old_event) = self.get_event_by_id(&event.get_id()) {
-            old_event.remove();
+
+    pub fn update_events(&self, events: Vec<FullCalendarEvent>) {
+        self.clear_events();
+        for event in events {
+            self.add_calendar_event(event);
```
Comment (42Pupusas) [inline src/widgets/full_calendar/full_calendar.rs]: please no unused results for `add_callentdar_events`

**Review by 42Pupusas — CHANGES_REQUESTED** [review]
Bindings and components look great, a few small details to review:

**Comment by 42Pupusas** [comment]
Also pushed some changes to the `browser_api` helpers today so please merge those changes in as well

File: src/widgets/full_calendar/full_calendar.rs (hunk 3 above)
Comment (rahat2134) [reply src/widgets/full_calendar/full_calendar.rs]: Done

File: src/widgets/full_calendar/component.rs (hunk 2 above)
Comment (rahat2134) [reply src/widgets/full_calendar/component.rs]: Done

File: src/widgets/full_calendar/test_component.rs (hunk 1 above)
Comment (rahat2134) [reply src/widgets/full_calendar/test_component.rs:18]: Done

**Comment by rahat2134** [comment]
@42Pupusas PTAL! Try to implement your feedback. I think everything is working fine.

File: src/widgets/full_calendar/component.rs (hunk 2 above)
Comment (42Pupusas) [reply src/widgets/full_calendar/component.rs]: sorry to nitpick but theres still one unused result:

image


im trying to make sure we get no warnings when using `cargo check` with default `rust-analyzer` options

Comment (rahat2134) [reply src/widgets/full_calendar/component.rs]: oh, I am really sorry for that, It just slipped from my eyes. I will push the changes. It is so silly of me :(

Comment (rahat2134) [reply src/widgets/full_calendar/component.rs]: @42Pupusas , I hope it's fine now. PTAL, If anything else you want please let me know. Thanks :)

**Comment by rahat2134** [comment]
@42Pupusas If this is good to go, can we get this merged?

**Review by 42Pupusas — APPROVED** [review]

**Pull request merged**